beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -270,28 +289,68 @@ func (s *Service) deleteBlockFromPendingQueue(slot uint64, b *ethpb.SignedBeacon
 		newBlks = append(newBlks, blk)
 	}
 	if len(newBlks) == 0 {
-		delete(s.slotToPendingBlocks, slot)
-		return
+		s.slotToPendingBlocks.Delete(slotToCacheKey(slot))
+		return nil
+	}
+
+	if err := s.slotToPendingBlocks.Replace(slotToCacheKey(slot), newBlks, gcache.DefaultExpiration); err != nil {
```

Comment:
Upong replacing it, we will reset the expiration time, so not the best side effect.

Reply:
I'm going to decrease exp time based on the remaining blocks in slot. I think this is a fair compromise without getting too deep

```go
	// Decrease exp itme in proportion to how many blocks are still in the cache for slot key.
	d := gcache.DefaultExpiration.Seconds() / float64(len(newBlks))
	if err := s.slotToPendingBlocks.Replace(slotToCacheKey(slot), newBlks, time.Duration(d)); err != nil {
		return err
	}
```